source/dubregistry/repositories/gitea.d

```diff
@@ -0,0 +1,284 @@
+/**
+	Copyright: © 2023 Sönke Ludwig
+	License: Subject to the terms of the GNU GPLv3 license, as written in the included LICENSE.txt file.
+	Authors: Sönke Ludwig
+*/
+module dubregistry.repositories.gitea;
+
+import dubregistry.cache;
+import dubregistry.dbcontroller : DbRepository;
+import dubregistry.repositories.repository;
+import std.string : startsWith;
+import std.typecons;
+import vibe.core.log;
+import vibe.core.stream;
+import vibe.data.json;
+import vibe.http.client : HTTPClientRequest;
+import vibe.inet.url;
+
+class GiteaRepositoryProvider : RepositoryProvider {
+	private {
+		string m_token;
+		string m_url;
+	}
+@safe:
+
+	private this(string token, string url)
+	{
+		m_token = token;
+		m_url = url.length ? url : "https://gitea.com/";
+	}
+
+	static void register(string token, string url)
+	{
+		auto h = new GiteaRepositoryProvider(token, url);
+		addRepositoryProvider("gitea", h);
+	}
+
+	bool parseRepositoryURL(URL url, out DbRepository repo)
+	@safe {
+		import std.algorithm.searching : endsWith;
+		import std.conv : to;
+
+		string host = url.host;
+		if (!url.startsWith(URL(m_url)) && !host.endsWith(".gitea.com") && host != "gitea.com" && host != "gitea")
```

Comment:
since this is only for private registries, I think we probably want to not have the gitea.com / gitea shortcuts here, or at most have only the shortcut here

Reply:
It's not only for private registries, although we can probably remove this anyway. I just took it over from the original URL parsing code.